src/d3a/models/market/blockchain_interface.py

```diff
@@ -101,8 +101,8 @@ def track_trade_event(self, trade):
             'trade_id': trade.id,
             'buyer': ALICE_STASH_ADDRESS,
             'seller': BOB_STASH_ADDRESS,
-            'amount': default_amount,
-            'rate': default_rate
+            'energy': trade.offer.energy * BC_NUM_FACTOR,
```

Comment:
```suggestion
            'energy': trade.offer.energy,
```
I think `BC_NUM_FACTOR ` could be skipped

Reply:
The reason why ETH is calculated in Wei is because Weis are always whole numbers, and ETH are not. So they use 10**18. Here they have the 10**12 factor, I would rather leave it to avoid any precision or rounding problems

Reply:
`energy` isn't now in  ETH or any currency like `AccountData` but its now simply in `u16`

Reply:
OK makes sense! Edited